ytmusicapi/ytmusic.py

```diff
@@ -87,12 +92,23 @@ def __init__(self,
         # value from https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/youtube.py#L502
         self.cookies = {'SOCS': 'CAI'}
         if self.auth is not None:
-            input_json = load_headers_file(self.auth)
-            self.input_dict = CaseInsensitiveDict(input_json)
-            self.input_dict['filepath'] = self.auth
+            if isinstance(self.auth, str):
+                input_json = load_headers_file(self.auth)
+                self.input_dict = CaseInsensitiveDict(input_json)
+                self.input_dict['filepath'] = self.auth
+
+            elif isinstance(self.auth, dict):
+                self.input_dict = self.auth
+
             self.is_oauth_auth = is_oauth(self.input_dict)
 
-        self.headers = prepare_headers(self._session, proxies, self.input_dict)
+        # use custom oauth client parameters if provided
+        if alt_oauth and isinstance(alt_oauth, dict):
+            self.alt_oauth = YTMusicOAuth(self._session, proxies, **alt_oauth)
```

Comment:
Here you are assuming that the `alt_oauth` dict contains the correct keys for the function, which are documented nowhere except the `YTMusicOAuth.__init__`. Not very transparent.

I suggest we change the `alt_oauth` argument to `YTMusic` to `oauth_credentials` of type `OAuthCredentials`, which can be a new dataclass located in `oauth.py`. It has fields `client_id` and `secret` with default values set to the existing constants. That should simplify things

Instead of calling `YTMusicOAuth` in `ytmusic.py`, just do it in `prepare_headers` by passing in the `OAuthCredentials`.

Reply:
I'm in favor of a dataclass implementation of OAuthCredentials.